test/File/RenameUploadTest.php

```diff
@@ -412,4 +412,50 @@ public function testReturnUnfiltered($input)
 
         $this->assertEquals($input, $filter($input));
     }
+
+    /**
+     * @see https://github.com/zendframework/zend-filter/issues/77
+     * @return void
+     */
+    public function testBackwordCompatibilityBreakFromRelease280ToRelease290()
+    {
+        $filter = new RenameUploadMock($this->targetPath);
+
+        // Emulate the output of \Zend\Http\Request::getFiles()->toArray()
+        $sapiSource = [
+            'tmp_name' => $this->sourceFile,
+            'name' => basename($this->targetFile),
+            'type' => 'text/plain',
+            'error' => \UPLOAD_ERR_OK,
+            'size' => 123,
+        ];
+
+        $sapiTarget = [
+            'tmp_name' => $this->targetPathFile,
+            'name' => basename($this->targetFile),
+            'type' => 'text/plain',
+            'error' => \UPLOAD_ERR_OK,
+            'size' => 123,
+        ];
+
+        // Check the result twice for the `alreadyFiltered` cache path
+        $this->assertEquals($sapiTarget, $filter($sapiSource));
+        $this->assertEquals($sapiTarget, $filter($sapiSource));
+    }
+
+    /**
+     * @see https://github.com/zendframework/zend-filter/issues/76
+     * @return void
+     */
+    public function testTargetSameAsSource()
```

Comment:
Similar comment here; the test name isn't telling me what the behavior being tested is. Let's try something along the lines of `testFilterReturnsFileDataVerbatimUnderSAPIWhenNameAndTmpNameDiffer()`.